tests/attr/test_class_summarizer.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+import torch
+
+from captum.attr import ClassSummarizer, CommonStats
+
+from .helpers.utils import BaseTest
+
+
+class Test(BaseTest):
+    def class_test(self, data, classes, x_sizes):
+        summarizer = ClassSummarizer(stats=CommonStats())
+        for x, y in data:
+            summarizer.update(x, y)
+
+        summ = summarizer.summary
+        self.assertIsNotNone(summ)
+        self.assertIsInstance(summ, list)
+        for s, size in zip(summ, x_sizes):
+            self.assertIsInstance(s, dict)
+            for key in s:
+                self.assertEqual(s[key].size(), size)
+
+        self.assertIsNotNone(summarizer.class_summaries)
+        all_classes = torch.zeros(len(classes))
+        class_summaries = summarizer.class_summaries
+        all_keys = set(class_summaries.keys())
+        for i, clazz in enumerate(classes):
+            self.assertTrue(clazz in class_summaries)
+            all_keys.remove(clazz)
+            all_classes[i] = 1
+
+            summ = class_summaries[clazz]
+            self.assertIsNotNone(summ)
+            self.assertIsInstance(summ, list)
+
+            for s, size in zip(summ, x_sizes):
+                self.assertIsInstance(s, dict)
+                for key in s:
+                    self.assertEqual(s[key].size(), size)
+
+        self.assertEqual(len(all_keys), 0)
+        self.assertEqual(all_classes.sum(), len(classes))
+
+    def test_classes(self):
+        sizes_to_test = [
+            ((1,),),
+            ((3, 2, 10, 3), (1,)),
+            ((20,)),
+        ]
+        classeses = [list(range(100)), ["%d" % i for i in range(100)]]
+        for batch_size in [None, 1, 4]:
+            for sizes, classes in zip(sizes_to_test, classeses):
+
+                def create_batch_labels(batch_idx):
+                    if batch_size is None:
+                        # batch_size = 1
+                        return classes[batch_idx]
+
+                    return classes[
+                        batch_idx * batch_size : (batch_idx + 1) * batch_size
+                    ]
+
+                sizes_plus_batch = sizes
+                if batch_size is not None:
+                    sizes_plus_batch = tuple((batch_size,) + si for si in sizes)
+
+                # we want to test each possible element
+                num_batches = len(classes)
+                if batch_size is not None:
+                    num_batches //= batch_size
+
+                data = [
+                    (
+                        tuple(torch.randn(si) for si in sizes_plus_batch),
+                        create_batch_labels(batch_idx),
+                    )
+                    for batch_idx in range(num_batches)
+                ]
+
+                with self.subTest(
```

Comment:
One thing to note is that pytest doesn't support subTest (which is what we use in CircleCI and sometimes when running locally). I think this will still run all the subtests sequentially as a single test, so it's not harmful to leave this, but might be good to keep in mind if you want the subTest information to show up in CircleCI failures / need them to run independently.